### DISCUSSION TITLE
I am starting a new software development company - I need help with designing it's GitHub account

### DISCUSSION THREAD

**Discussion by jumppard-lfc** [community/community#110865] [New to GitHub] [answered]
Hi guys,

I am planning to start a new software development company. 

Here is the company's use-case:
- There will be many customers. 
- For each customer, there will be delivered one or more projects. 
- My company will be hiring freelancing software developers (from for example fiverr.com) for specific tasks. I will always create some task and then I will hire a developer. 
- I need to restrict those developers to be able to see only tasks assigned to them. So they can't be able to see any other tasks.

There are lots of terms here in GitHub like Repositories / Projects / Collaborators / Organization / etc. I need to design a custom solution for my company needs. I am seeking help from you guys to help me with setting-up/design my GitHub account to accomplish above mentioned list of company "rules". 

Many thanks in advance for your help!

Kind regards,
Matej

**Comment by Aboodkaaboodkaaaa** [comment]
I’m





احصل على Outlook لـ iOS<https://aka.ms/o0ukef>
________________________________
من: jumppard-lfc ***@***.***>
تاريخ الإرسال: الاثنين, مارس ٤, ٢٠٢٤ ١٢:٠٤ م
إلى: community/community ***@***.***>
نسخة: Subscribed ***@***.***>
الموضوع: [GitHub Community] I am starting a new software development company - I need help with designing it's GitHub account (Discussion #110865)


Hi guys,

I am planning to start a new software development company.

Here is the company's use-case:

  *   There will be many customers.
  *   For each customer, there will be delivered one or more projects.
  *   My company will be hiring freelancing software developers (from for example fiverr.com) for specific tasks. I will always create some task and then I will hire a developer.
  *   I need to restrict those developers to be able to see only tasks assigned to them. So they can't be able to see any other tasks.

There are lots of terms here in GitHub like Repositories / Projects / Collaborators / Organization / etc. I need to design a custom solution for my company needs. I am seeking help from you guys to help me with setting-up/design my GitHub account to accomplish above mentioned list of company "rules".

Many thanks in advance for your help!

Kind regards,
Matej

**Comment by koeppel** [comment, marked as answer]
Hi,

you don't need a dedicated GitHub account for the company. You can simply create a so called Organization -> this will represent your company.
There you can create a Repository for each of the projects you want to deliver to your customers.

You can then add the freelancers GitHub account as Collaborators / Contributors to the Organization and setup Roles so they are only able to see the Repository for their given project.

If you want to organize the tasks in a more project-oriented way I'd suggest also adding a GitHub Project for each customer to the Organization. You can then define tasks directly in the GitHub Project or in the Repository via issues and link them to the overlaying Project. In this Project overview you can then assign different status, tags, start and end dates, priority and size to the given tasks.

I'd suggest taking a look at the documentations: 
- Organizations: https://docs.github.com/en/organizations/collaborating-with-groups-in-organizations/about-organizations.
- Projects: https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects

**Reply by jumppard-lfc** [reply]
Hi Janik,

many thanks for your insight and advice. **I like it!** I will definitely check the links you provided as well. Thanks again, it means a lot to me! M.

**Reply by koeppel** [reply]
Great you're welcome! Feel free to mark the answer as solving :)

**Comment by Nelda621** [comment]
That’s exciting! For designing your GitHub account, it's essential to focus on clear structure and easy navigation. Make sure your repositories are well-organized and have detailed README files to explain the purpose of each project. You could also use GitHub Pages for a professional website showcasing your work. Additionally, if you’re looking for tips on selecting the right software development partner, you can check out these helpful [tips for choosing a software development company](https://www.cleveroad.com/blog/how-to-choose-a-software-development-company/) to guide your decision-making.
